cohere/compass/utils.py

```diff
@@ -75,15 +93,19 @@ def open_document(document_path: str) -> CompassDocument:
 
 def scan_folder(
     folder_path: str,
-    allowed_extensions: Optional[List[str]] = None,
+    allowed_extensions: Optional[list[str]] = None,
     recursive: bool = False,
-) -> List[str]:
+) -> list[str]:
```

Comment:
Why are we changing this to `list`, i thought python preferred way is `typing.List`

Reply:
@ankush-cohere 

TL;DR - `list` is the recommended way starting from Python 3.9.

`typing.List` isn't actually the preferred. When type hinting was introduced in Python (in version 3.5), it was straight forward to support primitive types like `int`, `float` in a "natural way", i.e. using the types themselves without having to define a parallel type under the contrived `typing` module. However, generic types like `list`, `dict` was a bit more challenging to introduce, and hence the Python community introduced the `typing` model to address the challenges. But after [PEP 585](https://peps.python.org/pep-0585/), which was introduced in Python 3.9, it became possible to use the types directly, i.e. `list`, `dict`, etc., without having to define a parallel type in the `typing` module. Hence, types like `typing.Dict`, `typing.List`, etc., are now unnecessary and they are getting deprecated and will be removed in future versions of Python; see [the doc](https://docs.python.org/3/library/typing.html).